L1Trigger/L1TTrackMatch/plugins/L1TrackJetEmulatorProducer.cc

```diff
@@ -0,0 +1,452 @@
+// Original Author: Samuel Edwin Leigh, Tyler Wu,
+//                  Rutgers, the State University of New Jersey
+//
+// Improvements:      George Karathanasis,
+//                          georgios.karathanasis@cern.ch, CU Boulder
+//
+//         Created:  Wed, 01 Aug 2018 14:01:41 GMT
+//         Latest update: Nov 2022 (by GK)
+//
+// Track jets are clustered in a two-layer process, first by clustering in phi,
+// then by clustering in eta
+// Introduction to object (p10-13):
+// https://indico.cern.ch/event/791517/contributions/3341650/attachments/1818736/2973771/TrackBasedAlgos_L1TMadrid_MacDonald.pdf
+
+// L1T include files
+#include "DataFormats/L1TCorrelator/interface/TkJet.h"
+#include "DataFormats/L1TCorrelator/interface/TkJetFwd.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h"
+#include "DataFormats/L1Trigger/interface/TkJetWord.h"
+#include "DataFormats/L1Trigger/interface/VertexWord.h"
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+
+// system include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/CommonTopologies/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelGeomDetType.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "DataFormats/Common/interface/Ref.h"
+
+//own headers
+#include "L1Trigger/L1TTrackMatch/interface/L1TrackJetEmulationProducer.h"
```

Comment:
Out of curiosity, why is this `L1TrackJetEmulatorProducer` but it's header is `L1TrackJetEmulationProducer`? Could this be renamed to match header to .cc a little easier? Are these things connected?

Reply:
It was mostly to remind that this file is also used in simulation. But I agree might be weird choice so I changed the file and tested locally that it works